docs/design/automatic-updates.md

```diff
@@ -0,0 +1,38 @@
+
+# Storage Node automatic updates DesignDocument
+
+## Overview
+
+As more Storage Node Operators join the network we must ensure these nodes have a mechanism to automatically keep their nodes software up to date. If a Storage Node Operator fails to keep their node up to date with the minimum version required by satellites they will no longer be selected for upload or download requests.
+
+## Goals
+
+Create an automatic update system into the Storage Node software so that node operators do not have to worry about updating their nodes.
+Keep the Storage Nodes on the network up to date with the latest software to improve stability and performance. 
+Concerns
+
+A majority or substantial amount of nodes update their software at the same time causing instability in the network or temporary file lose.
+We must ensure the automatic update system we build into the Storage nodes works on ALL OS we support.
+
+## Scenarios
+
+
+## Design Overview
+
+When starting a storj node:
+* Main process starts an updater service which starts a (12 hour?) interval loop
```

Comment:
What if that service crashes or is stopped by something? Each OS will need special handling for it, probably the design doc should outline each OS separately.

Reply:
I don't understand how each os would be different

Reply:
e.g. API for Windows services https://docs.microsoft.com/en-us/windows/win32/services/services, this doesn't exist neither on Mac nor Linux.

Reply:
Anti-Virus for Windows requires properly signed executables, I assume we have that in place. Similarly, I think there were small details like when you name executable contains "Setup" and "Update", then it will trigger UAC.

Reply:
I don't know anything about this. Who can I talk to that knows about this and can work with me on rewriting the design doc

Reply:
I think a good start would be getting that hard list of operating systems, and creating a section for each one.  I agree with Egon that we should probably acknowledge "keeping the service running" as important as well.  The bummer is that if the updater tries to keep the storj node executable running, what keeps the updater running?  At some point, this necessarily becomes an operating system specific problem.